internal/monitors/metadata/hostmetadata/hostmetadata.go

```diff
@@ -0,0 +1,190 @@
+package hostmetadata
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"os"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/metadata/aws/ec2metadata"
+	"github.com/signalfx/golib/metadata/hostmetadata"
+	"github.com/signalfx/signalfx-agent/internal/core/config"
+	"github.com/signalfx/signalfx-agent/internal/monitors"
+	"github.com/signalfx/signalfx-agent/internal/monitors/metadata"
+	"github.com/signalfx/signalfx-agent/internal/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+const (
+	monitorType      = "host-metadata"
+	uptimeMetricName = "gauge.sf.host-plugin_uptime"
+)
+
+// MONITOR(host-metadata): This monitor collects metadata about a host.  It is
+// required for some views in SignalFx to operate.
+//
+// ```yaml
+// monitors:
+//   - type: host-metadata
+// ```
+//
+// In containerized environments host `/etc` and `/proc` may not be located
+// directly under the root path.  You can specify the path to `proc` and `etc`
+// using the monitor configurations `procFSPath` and `etcPath`
+//
+// ```yaml
+// monitors:
+//   - type: host-metadata
+//     procFSPath: "/hostfs/proc"
+//     etcPath: "/hostfs/etc"
+// ```
+
+// GAUGE(gauge.sf.host-plugin_uptime): The time this monitor has been running in
+// milliseconds.  Dimensions include `signalfx_agent`, `collectd`,
+// `kernel_release`, `kernel_version`
+
+var logger = log.WithFields(log.Fields{"monitorType": monitorType})
+
+func init() {
+	monitors.Register(monitorType, func() interface{} { return &Monitor{Monitor: &metadata.Monitor{}} }, &Config{})
+}
+
+// Config for this monitor
+type Config struct {
+	config.MonitorConfig `acceptsEndpoints:"false"`
+	// The path to the proc filesystem. Useful to override in containerized
+	// environments.
+	ProcFSPath string `yaml:"procFSPath" default:"/proc"`
+	// The path to the main host config dir. Useful to override in
+	// containerized environments.
+	EtcPath string `yaml:"etcPath" default:"/etc"`
+}
+
+// Monitor for host-metadata
+type Monitor struct {
+	*metadata.Monitor
+	startTime time.Time
+	cancel    func()
+}
+
+// Configure is the main function of the monitor, it will report host metadata
+// on a varied interval
+func (m *Monitor) Configure(conf *Config) error {
+	intervals := []time.Duration{
+		// 0-60 seconds
+		time.Duration(rand.Int63n(60)) * time.Second,
+		// 1 minute after the previous because sometimes pieces of metadata aren't available immediately on startup
+		time.Duration(60) * time.Second,
+		// 1 hour after the previous with some dither
+		time.Duration(rand.Int63n(60)+3600) * time.Second,
+		// 1 day after the previous with some dither
+		time.Duration(rand.Int63n(600)+86400) * time.Second,
+	}
+
+	// set plugin start time
+	m.startTime = time.Now()
+
+	// set HOST_PROC and HOST_ETC for gopsutil
+	if conf.ProcFSPath != "" {
+		if err := os.Setenv("HOST_PROC", conf.ProcFSPath); err != nil {
+			logger.Errorf("Error setting HOST_PROC env var %v", err)
```

Comment:
Shouldn't these errors cause the whole `Configure` method to return an error instead of just continuing on?